### PR TITLE
Add Ballista support to DataFusion CLI

### LINKED ISSUE #886: Update datafusion-cli to support Ballista, or implement new ballista-cli
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
The DataFusion CLI is very handy for demos and testing. It would be nice if we had a similar capability for Ballista.

**Describe the solution you'd like**
I would like a CLI that works with Ballista.

**Describe alternatives you've considered**
Add Ballista support to datafusion-cli, or create a new ballista-cli crate.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes #886 .

```
$ ./target/release/datafusion-cli --host localhost --port 50050
DataFusion CLI v4.0.0-SNAPSHOT

> create external table lineitem stored as parquet location '/mnt/tpch/parquet-sf100-partitioned/lineitem';
0 rows in set. Query took 0.015 seconds.

> select
    l_returnflag,
    l_linestatus,
    sum(l_quantity) as sum_qty,
    sum(l_extendedprice) as sum_base_price,
    sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
    avg(l_quantity) as avg_qty,
    avg(l_extendedprice) as avg_price,
    avg(l_discount) as avg_disc,
    count(*) as count_order
from
    lineitem
where
        l_shipdate <= date '1998-09-02'
group by
    l_returnflag,
    l_linestatus
order by
    l_returnflag,
    l_linestatus;
+--------------+--------------+------------+--------------------+--------------------+-------------------+--------------------+--------------------+----------------------+-------------+
| l_returnflag | l_linestatus | sum_qty    | sum_base_price     | sum_disc_price     | sum_charge        | avg_qty            | avg_price          | avg_disc             | count_order |
+--------------+--------------+------------+--------------------+--------------------+-------------------+--------------------+--------------------+----------------------+-------------+
| A            | F            | 3775127758 | 5660776097194.456  | 5377736398183.934  | 5592847429515.928 | 25.499370423275426 | 38236.116984304936 | 0.05000224353092895  | 148047881   |
| N            | F            | 98553062   | 147771098385.9801  | 140384965965.03497 | 145999793032.7757 | 25.501556956882876 | 38237.19938880453  | 0.049985284338054006 | 3864590     |
| N            | O            | 7436302959 | 11150725648169.863 | 10593195276359.283 | 11016932215670.58 | 25.500009433521782 | 38237.227663621445 | 0.0499979183499098   | 291619616   |
| R            | F            | 3775724970 | 5661603032745.348  | 5378513563915.405  | 5593662252666.917 | 25.50006628406532  | 38236.697258453016 | 0.050001304339654135 | 148067261   |
+--------------+--------------+------------+--------------------+--------------------+-------------------+--------------------+--------------------+----------------------+-------------+
4 rows in set. Query took 8.677 seconds.
```

 # Rationale for this change

Enables a SQL repl for Ballista with minimal effort.

# What changes are included in this PR?

datafusion-cli now has new `host` and `port` arguments

# Are there any user-facing changes?

No

### CODE REVIEW

**Review by houqp — APPROVED** [review]

**Comment by houqp** [comment]
looks like code blocks in readme needs to be marked with `no_run`

**Comment by houqp** [comment]
my bad, `no_run` still compiles the code, we would need to use `ignore` here :(

File: datafusion-cli/src/main.rs [hunk 1]
```diff
@@ -105,31 +139,43 @@ pub async fn main() {
         execution_config = execution_config.with_batch_size(batch_size);
     };
 
+    let ctx: Result<Context> = match (host, port) {
+        (Some(h), Some(p)) => {
+            let config: BallistaConfig = BallistaConfig::new()
+                .map_err(|e| DataFusionError::Execution(format!("{:?}", e)))?;
+            Ok(Context::Remote(BallistaContext::remote(h, p, &config)))
+        }
+        _ => Ok(Context::Local(ExecutionContext::with_config(
```
Comment (alamb) [inline datafusion-cli/src/main.rs:148]: 👍  perfect

**Review by alamb — APPROVED** [review]
I looked at the code and it looks good to me. I think we have reasonable CI test coverage for the `datafusion-cli` (as it is used in the python integration test suite that @Jimexist  started). 

Thanks @andygrove

**Comment by Jimexist** [comment]
Sorry that I've been quite busy these days. Will review by end of the week and also see what else i can contribute to. This is exciting!

**Pull request merged**